template/docs/conf.py.jinja

```diff
@@ -73,6 +73,34 @@ napoleon_type_aliases = {
 typehints_defaults = 'comma'
 typehints_use_rtype = False
 
+_typehint_aliases = {
+            'scipp._scipp.core.DataArray': 'scipp.DataArray',
+            'scipp._scipp.core.DataGroup': 'scipp.DataGroup',
+            'scipp._scipp.core.Dataset': 'scipp.Dataset',
+            'scipp._scipp.core.Variable': 'scipp.Variable',
+        }
+
+
+def typehints_formatter(annotation, config: sphinx.config.Config) -> Optional[str]:
+    """Format typehints with improved NewType handling."""
+    _  = config
+    # TODO Switch to isinstance(key, NewType) once our minimum is Python 3.10
+    # Note that we cannot pass mypy in Python<3.10 since NewType is not a type.
+    if (supertype := getattr(annotation, '__supertype__', None)) is not None:
+        projectname = '{% if namespace_package %}{{namespace_package}}.{% endif %}{{ projectname.removeprefix(namespace_package) }}'
+
+        if not annotation.__module__.startswith(projectname):
+            return None
```

Comment:
Why do nothing in this case? What if we depend on another package that uses aliases?

Reply:
I was cautious because I don't know other uses of `NewType`. And I thought the sphinx people must have a reason for their solution. But I can't think of a case right now where this would break something.

Reply:
Ok, let us leave it for now!